### ISSUE TITLE
What are the next seq2seqSharp examples?

### ISSUE THREAD

**Issue by GeorgeS2019** [zhongkaifu/Seq2SeqSharp#22]
Here are the categories of HuggingFace transformer examples

<img src="https://user-images.githubusercontent.com/49812372/120937669-09ee9680-c70f-11eb-8885-78cd4adc9011.png" alt="Girl in a jacket" width="200" height="200">

It seems you have started to build more .NET seq2seqSharp transformer examples (e.g. Text Classification). Perhaps the HuggingFace example categories could guide what could be the next seq2seqSharp examples the .NET community needs :-)

**Comment by zhongkaifu** [comment]
Hi @GeorgeS2019,

Thanks for your suggestion. I would rather like you or Seq2SeqSharp users or who is interested in it to vote which is the next example they would like to have. :)

In addition, for some examples, such as summarization, q&a, Seq2SeqSharp already support them in framework and code level, the only thing needed to build these examples are data.

Thanks
Zhongkai Fu

**Comment by GeorgeS2019** [comment]
@zhongkaifu   .NET ML communities are converging just like the path to .NET6.

Through TorchSharp, a strategy to "Convert" trained network of pyTorch to TorchSharp .NET friendly model data has been developed. 

Perhaps this could speed up having more Seq2SeqSharp transformer examples with pretrained networks based on HuggingFace transformer models based on PyTorch.

@zhongkaifu  if you want to grant me a Christmas gift, please focus on **Text generation e.g. GPT-2**

We already have a .[NET GPT-2 version with c# BPE tokenizer](https://github.com/xamarin/TorchSharp/issues/265)

.

**Comment by zhongkaifu** [comment]
yeah~ In fact, Seq2SeqSharp is already able to train GPT-2 model in framework level (Some code for randomly token sampling and masking needs to be implemented, but they are pretty simple. :) ), however, because pre-trained models don't work very well in my current works, (very long sequence), I'm not currently focusing on this part.  But if anyone is interested in it, it would be really appreciated. :) 

For Seq2SeqSharp, the next thing I may invest many time to do is to support ONNX which can break barrier between Seq2SeqSharp and other frameworks.

**Comment by zhongkaifu** [comment]
A standalone GPT decoder and console tool is here:  https://github.com/zhongkaifu/Seq2SeqSharp/commit/7fb859e418dbf8d7261660d144c0c81ddd9334ad
